presto-docs/src/main/sphinx/spill/overview.rst

```diff
@@ -0,0 +1,85 @@
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Overview
+--------
+
+In the case of memory intensive operations, Presto allows offloading
+intermediate operator results to disk. The goal of this mechanism is to
+enable execution of queries that require amounts of memory exceeding per query
+or per node limits.
+
+The mechanism is similar to OS level page swapping. However, it is
+implemented on the application level to address specific needs of Presto.
+
+Properties related to spilling are described in properties section
+:ref:`tuning-spilling`.
+
+
+Memory management and spill
+---------------------------
+
+By default, Presto manages memory needed by query runners manually to guarantee
+fairness in allocating memory to queries and prevent saturation of memory by a
+single query (could result in failure of other queries). In the basic case,
+if the memory requested by the query execution exceeds session properties
+``query_max_memory`` or ``query_max_memory_per_node``, the query is killed. This
+strategy is efficient when there is a lot of small queries in the cluster, but
+leads to killing large queries that don't stay within the limits.
+
+To overcome this inefficiency the concept of revocable memory was introduced.
+A query can request memory that does not count toward the limits, but this memory
+can be revoked by the memory manager at any time. To handle revoking query runner
```

Comment:
Suggest: When memory is revoked, the query runner spills intermediate data from memory to disk and continues to process it later.